routers/web/repo/setting/setting.go

```diff
@@ -119,867 +117,965 @@ func SettingsPost(ctx *context.Context) {
 	ctx.Data["SigningSettings"] = setting.Repository.Signing
 	ctx.Data["IsRepoIndexerEnabled"] = setting.Indexer.RepoIndexerEnabled
 
-	repo := ctx.Repo.Repository
-
 	switch ctx.FormString("action") {
 	case "update":
-		if ctx.HasError() {
-			ctx.HTML(http.StatusOK, tplSettingsOptions)
-			return
-		}
+		handleSettingsPostUpdate(ctx)
```

Comment:
Looks like it's better to move these changes to another PR.

Reply:
ps: this refactoring is related, to make it easier to make related frontend changes.

If we do not put it here, it will also be in "next PR", otherwise there will be one more PR, too many PRs.